### ISSUE TITLE
Add `trace` method to servicesHost

### ISSUE THREAD

**Issue by onigoetz** [TypeStrong/ts-loader#691]
### Expected Behaviour

Adding `"traceResolution": true,` to tsconfig.json should display logs of how imported modules are resolved. It's very useful for debugging.

### Actual Behaviour

Setting `traceResolution` to true makes no difference in the output. The cause is that the declared ServicesHost doesn't provide a `trace` method.

Can we add a method that will output those informations ?

Normal output is like:

```
Module 'http' was resolved as locally declared ambient module in file '/home/sgoetz/workspace/foundation/ts-project/server-war/src/main/frontend/node_modules/@types/node/index.d.ts'.
```

### Steps to Reproduce the Problem

Add  `"traceResolution": true,` to your `compilerOptions` in `tsconfig.json` and see that nothing is different in your build.

### Location of a Minimal Repository that Demonstrates the Issue.

I don't have one, but it works on any project using ts-loader

**Comment by flyon** [comment]
yes, right now I have a module that cannot be found even though its installed under @types, would like to debug with traceResolution, but neither in tsconfig.json nor through ts-loader -> compilerOptions overwrite works

**Comment by johnnyreilly** [comment]
I'm open to this being added. Would you like to submit a speculative PR and we can iterate on that?

**Comment by onigoetz** [comment]
Sure I can do that

**Comment by onigoetz** [comment]
Closing now that this was merged in #721
